src/main/java/networkbook/logic/commands/AddCommand.java

```diff
@@ -96,43 +100,23 @@ private Person addInfoToPerson(Person personToAddInfo, EditPersonDescriptor edit
-
-        return new Person(updatedName, updatedPhone, updatedEmails, updatedAddress, updatedTags, updatedPriority);
-    }
-
-    // TODO: for non-unique fields, change respective model to use list and append to the list
-    // TODO: now it just replaces the old value
-
-    private Phone addPhone(Person personToAddInfo, EditPersonDescriptor editPersonDescriptor) {
-        return editPersonDescriptor.getPhone().orElse(personToAddInfo.getPhone());
-    }
-
-    private UniqueList<Email> addEmails(Person personToAddInfo, EditPersonDescriptor editPersonDescriptor) {
-        return editPersonDescriptor.getEmails().orElse(personToAddInfo.getEmails());
-    }
-
-    private Address addAddress(Person personToAddInfo, EditPersonDescriptor editPersonDescriptor) {
-        return editPersonDescriptor.getAddress().orElse(personToAddInfo.getAddress());
-    }
-
-    private Set<Tag> addTags(Person personToAddInfo, EditPersonDescriptor editPersonDescriptor) {
-        return editPersonDescriptor.getTags().orElse(personToAddInfo.getTags());
-    }
-
-    private Priority addPriority(Person personToAddInfo, EditPersonDescriptor editPersonDescriptor)
-            throws CommandException {
-        Optional<Priority> oldPriority = personToAddInfo.getPriority();
-        Optional<Priority> newPriority = editPersonDescriptor.getPriority();
-        if (oldPriority.isPresent() && newPriority.isPresent()) {
-            throw new CommandException(MESSAGE_MULTIPLE_UNIQUE_FIELD);
-        }
-        return newPriority.orElse(oldPriority.orElse(null));
+        // TODO: for non-unique fields, change respective model to use list and append to the list
+        // TODO: now it just replaces the old value
+        Name updatedName = editPersonDescriptor.getName().orElse(personToAddInfo.getName());
+        Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToAddInfo.getPhone());
+        UniqueList<Email> updatedEmails = editPersonDescriptor.getEmails().orElse(personToAddInfo.getEmails());
+        WebLink updatedWebLink = editPersonDescriptor.getWebLink().orElse(personToAddInfo.getWebLink());
+        GraduatingYear updatedGraduatingYear = editPersonDescriptor.getGraduatingYear()
+                .orElse(personToAddInfo.getGraduatingYear());
+        Course updatedCourse = editPersonDescriptor.getCourse().orElse(personToAddInfo.getCourse());
+        Specialisation updatedSpecialisation = editPersonDescriptor.getSpecialisation()
+                .orElse(personToAddInfo.getSpecialisation());
+        Set<Tag> updatedTags = editPersonDescriptor.getTags().orElse(personToAddInfo.getTags());
+        Priority updatedPriority = editPersonDescriptor.getPriority().orElse(personToAddInfo.getPriority()
+                .orElse(null));
+
+        return new Person(updatedName, updatedPhone, updatedEmails, updatedWebLink, updatedGraduatingYear,
+                updatedCourse, updatedSpecialisation, updatedTags, updatedPriority);
```

Comment:
Please keep the old method structure so that complex list handling can be done separately in its own method.

Reply:
Done!